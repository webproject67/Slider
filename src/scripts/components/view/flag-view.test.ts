import FlagView from './flag-view';
import Model from '../model/model';

const main = document.createElement('div');
main.id = 'banana';

const state = {
  flag: true,
  from: -10000,
  fromPercent: 0,
  max: 100,
  min: 0,
  progress: true,
  range: 'one',
  draft: 0,
  start: 1,
  configuring: 0,
  scale: true,
  step: 1,
  to: -10000,
  toPercent: 100,
  view: 'horizontal',
};

const model = new Model(main, state);
const flagView = new FlagView();

describe('snapshot', () => {
  test('renders correctly element1', () => {
    expect(new FlagView().getElement(model)).toMatchSnapshot();
  });

  test('renders correctly element2', () => {
    model.setValue(['range'], ['range']);
    expect(new FlagView().getElement(model)).toMatchSnapshot();
  });

  test('renders correctly element3', () => {
    model.setValue(['from'], [10]);
    expect(new FlagView().getElement(model)).toMatchSnapshot();
  });

  test('renders correctly element4', () => {
    model.setValue(['from', 'view'], [-10000, 'vertical']);
    expect(new FlagView().getElement(model)).toMatchSnapshot();
  });

  test('renders correctly element5', () => {
    model.setValue(['from', 'to'], [10, 90]);
    expect(new FlagView().getElement(model)).toMatchSnapshot();
  });
});

describe('handleFlagMouseDown', () => {
  test('spyOn flag mouseDown', () => {
    const somethingSpy = jest.spyOn(flagView, 'handleFlagMouseDown');
    let evt: any;
    flagView.handleFlagMouseDown(model, evt);
    expect(somethingSpy).toHaveBeenCalledTimes(1);
  });
});
