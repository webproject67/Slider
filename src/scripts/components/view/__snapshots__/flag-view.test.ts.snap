// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot renders correctly element1 1`] = `
<div
  class="slider__flags"
>
  
      
      
  <span
    class="slider__flag slider__flag_maximum"
    style="left:100%"
  >
    100
  </span>
  
    
</div>
`;

exports[`snapshot renders correctly element2 1`] = `
<div
  class="slider__flags"
>
  
      
  <span
    class="slider__flag slider__flag_minimum"
    style="left:0%"
  >
    0
  </span>
  
      
  <span
    class="slider__flag slider__flag_maximum"
    style="left:100%"
  >
    100
  </span>
  
    
</div>
`;

exports[`snapshot renders correctly element3 1`] = `
<div
  class="slider__flags"
>
  
      
  <span
    class="slider__flag slider__flag_minimum"
    style="left:0%"
  >
    10
  </span>
  
      
  <span
    class="slider__flag slider__flag_maximum"
    style="left:100%"
  >
    100
  </span>
  
    
</div>
`;

exports[`snapshot renders correctly element4 1`] = `
<div
  class="slider__flags"
>
  
      
  <span
    class="slider__flag slider__flag-vertical slider__flag-vertical_minimum"
    style="top:0%"
  >
    0
  </span>
  
      
  <span
    class="slider__flag slider__flag-vertical slider__flag-vertical_maximum"
    style="top:100%"
  >
    100
  </span>
  
    
</div>
`;

exports[`snapshot renders correctly element5 1`] = `
<div
  class="slider__flags"
>
  
      
  <span
    class="slider__flag slider__flag-vertical slider__flag-vertical_minimum"
    style="top:0%"
  >
    10
  </span>
  
      
  <span
    class="slider__flag slider__flag-vertical slider__flag-vertical_maximum"
    style="top:100%"
  >
    90
  </span>
  
    
</div>
`;
